embassy-stm32/src/gpio.rs

```diff
@@ -99,6 +129,10 @@ impl<'d, T: Pin> Output<'d, T> {
             let n = pin.pin() as usize;
             r.pupdr().modify(|w| w.set_pupdr(n, vals::Pupdr::FLOATING));
             r.moder().modify(|w| w.set_moder(n, vals::Moder::OUTPUT));
+            if open_drain {
```

Comment:
if it's not openrain you still need to do the modify to set it as not opendrain, otherwise it stays if the pin has been previously used as opendrain

Reply:
I see you're returning it to PUSHPULL on drop... Other stuff like i2c might still leave it in OPENDRAIN, I think it's safer to always set it anyway.